qcodes/instrument/base.py

```diff
@@ -625,6 +625,8 @@ def find_instrument(cls, name: str,
             TypeError: If a specific class was requested but a different
                 type was found.
```

Comment:
Update docstring to document that a KeyError is raised if not found

Reply:
it's already documented:
> KeyError: If no instrument of that name was found, or if its reference is invalid (dead).